clients/src/test/java/org/apache/kafka/common/utils/MockTime.java

```diff
@@ -20,13 +20,20 @@
 public class MockTime implements Time {
 
     private long nanos = 0;
+    private long autoTickMs = 0;
 
     public MockTime() {
         this.nanos = System.nanoTime();
     }
 
+    public MockTime(long autoTickMs) {
+        this.nanos = System.nanoTime();
+        this.autoTickMs = autoTickMs;
```

Comment:
My reasoning is that the tests that expect the clock to auto-tick would be affected if the code under test changed from `milliseconds` to `nanoseconds`. Am I missing something? And is there a reason to only auto-tick `milliseconds`? `FastClock` is an interesting idea, it seems to depend less on how often `milliseconds` is invoked by the code under test. It seems more realistic too (in a sense, it's like reducing all timeout values by a multiplier).